File: python/hash_table.py
menu = {}

menu["coca cola"] = 0.89
menu["pepsi"] = 0.79
menu["virgin cola"] = 0.69

print(menu)
print(menu["pepsi"])


guest_list = {}


def check_name(name):
    if guest_list.get(name):
        print(name + ' has been used to enter already - NO ENTRY!')
    else:
        guest_list[name] = True
        print('Come right in ' + name + '!')


check_name("david")
check_name("lola")
check_name("david")

File: javascript/hashTable.js
"use strict";

const menu = {};

menu.coca_cola = 0.89;
menu.pepsi = 0.79;
menu.virgin_cola = 0.69;

console.log(menu);
console.log(menu.pepsi);

guestList = {};

const checkName = (name) => {
  if (guestList[name]) {
    console.log(`${name} has been used to enter already - NO ENTRY!`);
  } else {
    guestList[name] = true;
    console.log(`Come right in ${name}!`);
  }
};

checkName("david");
checkName("lola");
checkName("david");
